packages/core/src/components/Tooltip/Tooltip.jsx

```diff
@@ -0,0 +1,214 @@
+import { Manager, Popper, Reference } from 'react-popper';
+import Transition, {
+  ENTERED,
+  ENTERING,
+  EXITING
+} from 'react-transition-group/Transition';
+import Button from '../Button/Button';
+import FocusTrap from 'focus-trap-react';
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import TooltipIcon from './TooltipIcon';
+import classNames from 'classnames';
+
+const transitionDuration = 300;
+const defaultTransitionStyle = {
+  transition: `opacity ${transitionDuration}ms`,
+  opacity: 0
+};
+const transitionStyles = {
+  [ENTERING]: { opacity: 0 },
+  [ENTERED]: { opacity: 1 },
+  [EXITING]: { opacity: 0 }
+};
+
+const TOOLTIP_OFFSET_OPT = '5, 5, 5, 5';
+const TOOLTIP_OFFSET = 5;
+
+class Tooltip extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      showTooltip: false,
+      focusEventTriggered: false
+    };
+  }
+
+  handleFocus() {
+    this.setState({ showTooltip: true, focusEventTriggered: true });
+  }
+
+  hideTooltip() {
+    this.setState({ showTooltip: false, focusEventTriggered: false });
+  }
+
+  showTooltip() {
+    this.setState({ showTooltip: true });
+  }
+
+  render() {
+    const body = document.querySelector('body');
+    return (
+      <Manager>
+        <Reference>
+          {({ ref }) => (
+            <button
```

Comment:
Have you thought about allowing this component to support alternate use cases for the trigger element (like `input`, `a`, or `span` elements)?

Reply:
The learn team uses [this tooltip](
https://gyazo.com/b67509066b6baab737067bd399122716) so I think its a good idea! However, I'm not sure whats the best way to implement these alternate use cases. 

I feel like allowing entirely custom tooltip triggers might not be ideal, i.e. I'm not sure if we want `input` or `a` elements that act like tooltips. Maybe instead of allowing custom markup we can just provide an optional `triggerLabel` prop that will replace the icon?

Reply:
The screenshot you shared seems pretty similar to the design QPP came up with for their glossary tooltips as well (dotted-underlined text with tooltip dialog on hover).  Between the "icon tip" and the "label tip", those seem like the only 2 scenarios we've had so far for tooltips and tooltip dialogs so baking those two cases into the component props makes sense.